hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestDatanodeManager.java

```diff
@@ -138,6 +138,35 @@ public void testNumVersionsCorrectAfterReregister()
         mapToCheck.get("version1").intValue(), 1);
   }
 
+  /**
+   * This test checks that if a node is re-registered with a different ip, its
+   * host2DatanodeMap is correctly updated with the new ip.
+   */
+  @Test
```

Comment:
It looks like there's a couple caches like this one. Is there equivalent test coverage for all the registered DN state that is updated on registration? Can you add coverage for any of the other states that are missing?

Reply:
When I checked the code, I think there is another one Network map, its code is what I followed for this hostToDatanodeMap change. Let me check, would rather avoid adding more testing cases for code which is already there.

Reply:
Yeah, the NetworkTopology map is another map. It is quite complicated to add a unittest for it. I am going to leave to  hdfs experts to decide if a unittest is needed.